File: cp1Endsem/c.cpp
#include <bits/stdc++.h>
using namespace std;
#define int long long int 
struct Range {
    int x, y, index;
};

bool compareByStart(const Range &a, const Range &b) {
    if (a.x == b.x) return a.y > b.y;
    return a.x < b.x;
}

bool compareByEnd(const Range &a, const Range &b) {
    if (a.y == b.y) return a.x > b.x;
    return a.y < b.y;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);

    int n;
    cin >> n;
    vector<Range> ranges(n);
    for (int i = 0; i < n; ++i) {
        cin >> ranges[i].x;
        ranges[i].index = i;
    }
    for (int i = 0; i < n; ++i) {
        cin >> ranges[i].y;
    }

    vector<int> ress(n, 0);
    vector<int> containedByResult(n, 0);

    sort(ranges.begin(), ranges.end(), compareByStart);

    int maxY = INT_MIN;
    for (int i = 0; i < n; ++i) {
        if (ranges[i].y < maxY) {
            ress[ranges[i].index] = 1;
        }
        maxY = max(maxY, ranges[i].y);
    }

    sort(ranges.begin(), ranges.end(), compareByEnd);

    int minY = INT_MAX;
    for (int i = n - 1; i >= 0; --i) {
        if (ranges[i].y > minY) {
            containedByResult[ranges[i].index] = 1;
        }
        minY = min(minY, ranges[i].y);
    }

    for (int i = 0; i < n; ++i) {
        cout << ress[i] << " ";
    }
    cout << "\n";
    for (int i = 0; i < n; ++i) {
        cout << containedByResult[i] << " ";
    }
    cout << "\n";

    return 0;
}

File: cp1Endsem/c.py
class Range:
    def __init__(self, x, y, index):
        self.x = x
        self.y = y
        self.index = index

def compare_by_start(a, b):
    if a.x == b.x:
        return b.y - a.y
    return a.x - b.x

def compare_by_end(a, b):
    if a.y == b.y:
        return b.x - a.x
    return a.y - b.y

def main():
    import sys
    input = sys.stdin.read
    data = input().split()
    
    n = int(data[0])
    ranges = []
    index = 1
    for i in range(n):
        x = int(data[index])
        ranges.append(Range(x, 0, i))
        index += 1
    for i in range(n):
        ranges[i].y = int(data[index])
        index += 1
    
    ress = [0] * n
    contained_by_result = [0] * n
    
    ranges.sort(key=lambda r: (r.x, -r.y))
    
    max_y = float('-inf')
    for i in range(n):
        if ranges[i].y < max_y:
            ress[ranges[i].index] = 1
        max_y = max(max_y, ranges[i].y)
    
    ranges.sort(key=lambda r: (r.y, -r.x))
    
    min_y = float('inf')
    for i in range(n-1, -1, -1):
        if ranges[i].y > min_y:
            contained_by_result[ranges[i].index] = 1
        min_y = min(min_y, ranges[i].y)
    
    print(' '.join(map(str, ress)))
    print(' '.join(map(str, contained_by_result)))

if __name__ == "__main__":
    main()